### PR TITLE
configure local storage static provisioner

### PR DESCRIPTION
**Pull request by gnufied**
This PR adds ci-operator config and job for https://github.com/openshift/sig-storage-local-static-provisioner

Looks like upstream `Dockerfile` however depends on debian image and runs `apt-get` as in Dockerfile.

### CODE REVIEW

File: ci-operator/config/openshift/sig-storage-local-static-provisioner/openshift-local-storage-static-provisioner-master.yaml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  base:
+    name: "4.2"
+    namespace: ocp
+    tag: base
+build_root:
+  image_stream_tag:
+    cluster: https://api.ci.openshift.org
+    name: release
+    namespace: openshift
+    tag: golang-1.10
```
Comment (jsafrane) [inline ci-operator/config/openshift/sig-storage-local-static-provisioner/openshift-local-storage-static-provisioner-master.yaml]: you may want 1.11 (at least Kubernetes uses 1.11 in 4.2)

File: ci-operator/config/openshift/sig-storage-local-static-provisioner/openshift-local-storage-static-provisioner-release-4.1.yaml [hunk 2]
```diff
@@ -0,0 +1,39 @@
+base_images:
+  base:
+    name: "4.1"
```
Comment (jsafrane) [inline ci-operator/config/openshift/sig-storage-local-static-provisioner/openshift-local-storage-static-provisioner-release-4.1.yaml]: Do we want to ship it in 4.1? I'd remove the file if not.

**Comment by gnufied** [comment]
pushed release branches in the source repository.
/retest

**Comment by jsafrane** [comment]
You also want to add your repo to tide `queries:` so it merges PRs with lgmt label.  It's somewhere around this: 
https://github.com/openshift/release/blob/b07bb2fe2311206c8c16ad70485461a3e3f7cc07/cluster/ci/config/prow/config.yaml#L184

(learned the hard way today :-)

**Comment by gnufied** [comment]
@jsafrane fixed!

**Comment by gnufied** [comment]
@petr-muller @stevekuznetsov can you guys take a look please?

File: ci-operator/config/openshift/sig-storage-local-static-provisioner/openshift-sig-storage-local-static-provisioner-release-4.3.yaml [hunk 3]
```diff
@@ -0,0 +1,39 @@
+base_images:
```
Comment (stevekuznetsov) [inline ci-operator/config/openshift/sig-storage-local-static-provisioner/openshift-sig-storage-local-static-provisioner-release-4.3.yaml]: Do you actually have all these branches that you actually want to test the same way?

**Comment by stevekuznetsov** [comment]
```
failed to determine job spec: failed to resolve --git-ref: 'git ls-remote https://github.com/openshift/local-storage-static-provisioner.git release-4.2' failed with 'exit status 128'
```

Are you adding jobs for branches on the repo that don' exist?

**Comment by gnufied** [comment]
@stevekuznetsov are you looking at old failures? Those branches do actually exist in target repo.

**Comment by stevekuznetsov** [comment]
Oh right, those are old failures in the comment above.

/lgtm

**Pull request merged**